### PR TITLE
Split things up so we've got better separation

### PR DESCRIPTION
**Pull request by matt-richardson**
# Background
 
The code in this plugin has got a little messy. We always intended to refactor it a bit.

# Results

This PR refactors the honeycomb/zipkin/custom services into their own separate places.
This is part 1 - there's still more to be done.

# How to review this PR

# Pre-requisites
- [ ] I have considered informing or consulting the right people
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: server/src/main/java/com/octopus/teamcity/opentelemetry/server/endpoints/custom/CustomOTELEndpointHandler.java [hunk 1]
```diff
@@ -52,6 +54,25 @@ public SetProjectConfigurationSettingsRequest GetSetProjectConfigurationSettings
...
+        model.put("otelEndpoint", params.get(PROPERTY_KEY_ENDPOINT));
+
+        var headers = new ArrayList<HeaderDto>();
+        params.forEach((k,v)->{
+            if (k.startsWith(PROPERTY_KEY_HEADERS)) {
+                var key = k.substring(PROPERTY_KEY_HEADERS.length());
+                key = key.substring(1, key.length() - 1);
+                var header = new HeaderDto(key,
+                        EncryptUtil.isScrambled(v) ? EncryptUtil.unscramble(v) : v,
+                        EncryptUtil.isScrambled(v) ? "password" : "plaintext");
```
Comment (gupta-kartik) [inline server/src/main/java/com/octopus/teamcity/opentelemetry/server/endpoints/custom/CustomOTELEndpointHandler.java:68]: This seems wrong. I do understand that this is pre-existing code. So, ignore if it is doing what it is supposed to.

**Review by gupta-kartik — APPROVED** [review]
Code changes LGTM. Left a few comments for your consideration.

File: README.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ Under `Appenders`:
 
 Under `Loggers`:
 ```xml
-    <Logger name="com.octopus.teamcity.opentelemetry" level="DEBUG">
+    <Logger name="com.octopus.teamcity.opentelemetry" level="DEBUG" additivity="false">
```
[the remark this answers is not in the document]
Comment (matt-richardson) [reply README.md:57]: I think it goes "please ignore other loggers". The docs were a bit unclear.

File: server/src/main/java/com/octopus/teamcity/opentelemetry/server/endpoints/custom/CustomOTELEndpointHandler.java (hunk 1 above)
Comment (matt-richardson) [reply server/src/main/java/com/octopus/teamcity/opentelemetry/server/endpoints/custom/CustomOTELEndpointHandler.java:68]: Yeah, pre-existing, and works. 🙈

File: server/src/main/java/com/octopus/teamcity/opentelemetry/server/ProjectConfigurationTab.java [hunk 3]
```diff
@@ -54,7 +60,12 @@ public void fillModel(@NotNull Map<String, Object> model, @NotNull HttpServletRe
                     model.put("isOverridden", true);
                     var projectId = features.stream().skip(1).findFirst().get().getProjectId();
                     var sourceProject = projectManager.findProjectByExternalId(projectId);
-                    model.put("overwritesInheritedFromProjectName", sourceProject.getName());
+                    //todo: i think we should be referring to internal id here
```
[the remark this answers is not in the document]
Comment (matt-richardson) [reply server/src/main/java/com/octopus/teamcity/opentelemetry/server/ProjectConfigurationTab.java:63]: Nah - it's pre-existing. Unrelated to this PR, but probably something we should tackle.

**Pull request merged**